skill-tree/src/test/java/com/RDS/skilltree/integration/EndorsementsIntegrationTests.java

```diff
@@ -0,0 +1,520 @@
+    }
+
+    @Test
+    @DisplayName("Return 200 on endorsements creation")
+    public void testAPIReturns201_OnEndorsementCreation() {
+        UUID userId = user.getId();
+        UUID skillId = skill.getId();
+
+        EndorsementDRO endorsementDRO = new EndorsementDRO();
+        endorsementDRO.setUserId(userId);
+        endorsementDRO.setSkillId(skillId);
+        Response response =
+                given()
+                        .cookies(RestAPIHelper.getUserCookie())
+                        .contentType("application/json")
+                        .body(endorsementDRO)
+                        .post("/v1/endorsements");
+
+        response
+                .then()
+                .statusCode(201)
+                .contentType("application/json")
+                .body("data.user.firstName", equalTo("John"))
+                .body("data.skill.name", equalTo("Java"));
+    }
+
+    @Test
+    @DisplayName("Return 400 on endorsements userid null")
+    public void testAPIReturns400_OnEndorsementCreationUserIdNull() {
+
+        UUID skillId = skill.getId();
+
+        EndorsementDRO endorsementDRO = new EndorsementDRO();
+
+        endorsementDRO.setSkillId(skillId);
+        Response response =
+                given()
+                        .cookies(RestAPIHelper.getUserCookie())
+                        .contentType("application/json")
+                        .body(endorsementDRO)
+                        .post("/v1/endorsements");
+
+        response
+                .then()
+                .statusCode(400)
+                .contentType("application/json")
+                .body("data", equalTo(null))
+                .body("message", equalTo("user id cannot be null"));
+    }
+
+    @Test
+    @DisplayName("Return 400 on endorsements skillid null")
+    public void testAPIReturns400_OnEndorsementCreationSkillIdNull() {
+        UUID userId = user.getId();
+
+        EndorsementDRO endorsementDRO = new EndorsementDRO();
+        endorsementDRO.setUserId(userId);
+
+        Response response =
+                given()
+                        .cookies(RestAPIHelper.getUserCookie())
+                        .contentType("application/json")
+                        .body(endorsementDRO)
+                        .post("/v1/endorsements");
+
+        response
+                .then()
+                .statusCode(400)
+                .contentType("application/json")
+                .body("data", equalTo(null))
+                .body("message", equalTo("skill id cannot be null"));
+    }
+
+    @Test
+    @Disabled
+    @DisplayName("Return 200 on endorsements updation")
+    public void testAPIReturns200_OnEndorsementGivenId() {
+        String endorsementId = "e-1";
+        Response response =
+                given()
+                        .cookies(RestAPIHelper.getUserCookie())
+                        .pathParam("endorsementId", endorsementId)
+                        .patch("/endorsements");
```

Comment:
If this test checks complete updation of the Endorsement then shouldn't this be `put`?
Or does this check for partial updates?

Reply:
As discussed, this test existed before, seems like it is not doing updation as well. We might have to revisit that when we take up implementation of other methods of /endorsements